test/Microsoft.AspNet.Mvc.FunctionalTests/ActionResultTests.cs

```diff
@@ -2,30 +2,28 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
-using System.IO;
 using System.Net;
 using System.Net.Http;
 using System.Net.Http.Headers;
 using System.Text;
 using System.Threading.Tasks;
-using System.Xml.Serialization;
 using ActionResultsWebSite;
 using Microsoft.AspNet.Builder;
 using Microsoft.AspNet.TestHost;
+using Microsoft.Framework.DependencyInjection;
 using Xunit;
 
 namespace Microsoft.AspNet.Mvc.FunctionalTests
 {
     public class ActionResultTests
     {
-        private readonly IServiceProvider _provider = TestHelper.CreateServices("ActionResultsWebSite");
         private readonly Action<IApplicationBuilder> _app = new Startup().Configure;
-                
+
         [Fact]
         public async Task BadRequestResult_CanBeReturned()
         {
             // Arrange
-            var server = TestServer.Create(_provider, _app);
+            var server = TestServer.Create(_app, AddServices);
```

Comment:
nit: do you prefer this over just doing it inline?  

```
var service = TestService.Create(_app, services => TestHelper.AddServices(services, "ActionResultsWebSite"));
```

If I was going to have a private helper, I would do this and hide everything:
```
  var server = CreateTestServer();
```